src/test/java/gui/controllers/RoomControllerTest.java

```diff
@@ -0,0 +1,304 @@
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        Label label = new Label();
+        controller.setNumOfChestsOpened(label);
+
+        controller.updateChests(1);
+        assertTrue(label.getText().contains("1/3"));
+        controller.getProgress().getApiHandler().stopServer();
+    }
+
+    /**
+     * Test changeInformation method.
+     */
+    @Test
+    void changeInformationTest() {
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        Label label = new Label();
+        controller.setGameStatus(label);
+        camHandler.setBeginTime(0);
+        controller.setProgressBar(new GridPane());
+
+        controller.changeInformation(1);
+        assertTrue(label.getText().contains("started"));
+
+        controller.getProgress().getRoom().setTargetDuration(0);
+        controller.changeInformation(1);
+        assertTrue(label.getText().contains("ended"));
+
+        controller.getProgress().getApiHandler().stopServer();
+    }
+
+    /**
+     * Test updateWarning method.
+     */
+    @Test
+    void updateWarningPaneTest() {
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        Pane pane = new Pane();
+        Pane pane2 = new Pane();
+        pane.getChildren().add(new Pane());
+        pane.getChildren().add(new Pane());
+        pane.getChildren().add(pane2);
+        controller.setStatusPane(pane);
+
+        controller.updateWarningPane();
+        assertFalse(pane2.isVisible());
+
+        controller.getProgress().getApiHandler().stopServer();
+    }
+
+    /**
+     * Test changeTime method.
+     */
+    @Test
+    void changeTimeTest() {
+        PlatformImpl.startup(() -> {
+        });
+
+        RoomController controller = new RoomController();
+        CameraHandler camHandler = new CameraHandler();
+        camHandler.setBeginTime(1);
+        controller.setCameraHandler(camHandler);
+        controller.configure(testConfigFile);
+
+        GridPane pane = new GridPane();
+        pane.getChildren().add(new Pane());
+        controller.setProgressBar(pane);
+        controller.getProgress().updateProgress();
+        controller.fillProgress(2);
+        controller.changeTime(1);
+
+        List<Label> labels = controller.getChestTimeStampList();
+        assertEquals("00:00", labels.get(0).getText());
+        assertEquals("", labels.get(1).getText());
```

Comment:
Waarom is dit leeg? Moet de tijd niet oplopen?

Reply:
dat is de tijd van de tweede kist